exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StatusResources.java

```diff
@@ -112,7 +115,9 @@ public String getMetrics(@PathParam("hostname") String hostname) throws Exceptio
   private List<OptionWrapper> getSystemOptionsJSONHelper(boolean internal)
   {
     List<OptionWrapper> options = new LinkedList<>();
-    OptionManager optionManager = work.getContext().getOptionManager();
+    OptionManager optionManager = authEnabled.separateWorkspace()
```

Comment:
This is a complex bit of code which is repeated several times. Can it be factored out somewhere?